modules/scripts/startup-script/examples/install_ansible.sh

```diff
@@ -14,40 +14,117 @@
 # limitations under the License.
 
 apt_wait() {
-	while fuser /var/lib/dpkg/lock >/dev/null 2>&1; do
+	while fuser /var/lib/dpkg/lock 2>/dev/null 2>&1; do
 		echo "Sleeping for dpkg lock"
 		sleep 3
 	done
-	while fuser /var/lib/apt/lists/lock >/dev/null 2>&1; do
+	while fuser /var/lib/apt/lists/lock 2>/dev/null 2>&1; do
 		echo "Sleeping for apt lists lock"
 		sleep 3
 	done
 	if [ -f /var/log/unattended-upgrades/unattended-upgrades.log ]; then
 		echo "Sleeping until unattended-upgrades finishes"
-		while fuser /var/log/unattended-upgrades/unattended-upgrades.log >/dev/null 2>&1; do
+		while fuser /var/log/unattended-upgrades/unattended-upgrades.log 2>/dev/null 2>&1; do
 			sleep 3
 		done
 	fi
 }
 
-if [ ! -h /usr/bin/ansible-playbook ] || [ ! -f /usr/bin/ansible-playbook ]; then
-	if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ] || [ -f /etc/oracle-release ] || [ -f /etc/system-release ]; then
-		if [ ! -f /bin/pip ]; then
-			curl -Os https://bootstrap.pypa.io/pip/2.7/get-pip.py
-			/usr/bin/python get-pip.py
+# Installs any dependencies needed for python based on the OS
+install_python_deps() {
+	if [ -f /etc/debian_version ] || grep -qi ubuntu /etc/lsb-release 2>/dev/null ||
+		grep -qi ubuntu /etc/os-release 2>/dev/null; then
+		apt-get install -y python3-distutils
+	fi
+}
+
+# Gets the name of the python executable for python starting with python3, then
+# checking python. Sets the variable to an empty string if neither are found.
+get_python_path() {
+	python_path=""
+	if [ -f /bin/python3 ]; then
+		python_path="/bin/python3"
+	elif [ -f /bin/python ]; then
+		python_path="/bin/python"
```

Comment:
In the images I tested with, /bin/python3 was typically the source of /usr/bin/python3 - both existed. By looking for /bin/python{3,}, we are only looking for the system level python. Any other installation may be in user space and not something we want to interfere with.

Alternatively, this could be a variable, letting the user specify the correct python to use in the startup script module, defaulting to /bin/python3.

Reply:
I did a bit more testing, and one OS did in fact fail to match that pattern, ubuntu. For the sake of not adding significantly more complexity to handle that one case, I'm adding a 3rd fallback option for `which python3`. 

The changes has been added and pushed.